ui: Statements tracing - Diagnostics tab with data

koorosh opened the pull request:
Depends on: #45494, #45492, #45491, #45274 

- Add table to represent diagnostics data of statement
- Connect Statements view with statement selector to properly
display tab header (whether diagnostics data is present or not)

koorosh committed:
ui: (fix) Apply styles for navigation panes

Before, Text component styles were adjusted to style
links on side navigation panel. It was wrong because
Text component should not care about particular styling
of navigation pane.
Now, particular styles which relate to navigation panel
only is moved to SideNavigation component.
It is changed now because current implementation breaks
styles for Diagnostics panel.

Release note: None

koorosh committed:
ui: Link component

Link component is a wrapper on top of <a> element
is created to maintain consistent styles for links.

Release note: None

koorosh committed:
ui: Add Diagnostics view for Statements

- Adjust SummaryCard to provide external styles
- Add DiagnosticsView component which renders
view when no diagnostics available.
- Current implementation includes only view without
any logic connected to redux store.

Release note: None

koorosh committed:
ui: Fake endpoints for Statement tracing data

Note: this commit must not be merged in master branch
it is for demo purposes only.

It extends `getStatements` api response with diagnostics field
which contains tracing data info.

Release note: None

koorosh committed:
ui: Fake endpoints for Statement tracing data

Note: this commit must not be merged in master branch
it is for demo purposes only.

It extends `getStatements` api response with diagnostics field
which contains tracing data info.

Release note: None

koorosh committed:
ui: Add sagas to handle statement tracing actions

Statements sagas are implemented to be able request
statement diagnostics and retrieve when its ready.

This commit defines Actions for requesting diagnostics
and sagas which listens for those actions.

Also selectors are added to find statement by
statement id (query) which contains diagnostic data.

Release note: None

koorosh committed:
ui: Diagnostics tab with tracing records

- Add table to represent diagnostics data of statement
- Connect Statements view with statement selector to properly
display tab header (whether diagnostics data is present or not)

Release note (admin ui): Add Diagnostics tab for statement details page

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45495)
<!-- Reviewable:end -->

dhartunian committed:
DELETE!!! add API for accessing statement diagnostics

Added 3 new RPC endpoints under `_status` for manipulating Statement Diagnostics
Requests:

1. `CreateStatementDiagnosticsRequest`
Creates a new request for a given statement fingerprint

2. `StatementDiagnosticsRequests`
Fetches all statement diagnostics requests

3. `StatementDiagnostics`
Fetches a specific statement diagnostics request that has been completed and
includes the JSON payload of the trace

Basic implementations for these endpoints have been added to the
`apiReducers.tsx` and `api.ts` files for use by the Admin UI.

A minor bug where completed requests did not have `completedAt` timestamp set
was fixed.

Release note (<category, see below>): <what> <show> <why>

koorosh committed:
ui: Rename Link component to Anchor

Release note: None

koorosh committed:
ui: DEL! Stab for Statement Diagnostics API

Release note: None

koorosh committed:
ui: Integrate requestStatementDiagnostics API

Replace faked API with real implementation to
request statement diagnostics.

Release note: None

koorosh committed:
ui: Add selectDiagnosticRequestsByStatementFingerprint selector

Release note: None

koorosh committed:
ui: Connect Statement Diagnostics API

Before, Statements diagnostics view used faked API
to simulate responses from server.
Now redux API has changed to comply with actual
implementation of API.

Release note: None

koorosh committed:
ui: Connect Statement Diagnostics API

Before, Statements diagnostics view used faked API
to simulate responses from server.
Now redux API has changed to comply with actual
implementation of API.

Release note: None

koorosh committed:
ui: Connect Statement Diagnostics API wit diagnostics table component

Replacing faked API responses with real implementation required some
redesigns with following important changes:

- Requesting diagnosticsRequests initiated when DiagnosticsView is opened.
Then to request new diagnostics requests before existing data is invalidated,
we need to manually invalidate and then refresh data.

- Current backend API doesn't support Error state for requested data so only
"Ready", and "Waiting for query" statuses are possible.

Release note: None

koorosh committed:
ui: Add DownloadFile component

DownloadFile component accepts file name and content and
then initiates file download from client.
It works as a wrapper to catch onClick event on wrapped
component.

Release note: None

koorosh committed:
ui: Refactor Statement diagnostics action ()redux) names

Release note: None

koorosh committed:
ui: Add traces downloading on Statement Diagnostics tab

- Add ability to download json file with tracing data
- Refactor code to comply with changed backend API
- add `long` package because exposed interface for
getting statement diagnostics require diagnostics id
as Long type. Current library provides compatible
implementation of Long type with protobuf.js interfaces.

Release note: None

koorosh commented:
Outdated
